{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Investigating suburban micromoth diversity using DNA barcoding of malaise trap samples\n",
    "https://link-springer-com.ezproxy.gc.cuny.edu/article/10.1007%2Fs11252-016-0597-2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('>GMNWB040-14|Lyonetia clerkella|COI-5P\\r\\n', 'ATTTTATATTTTATTTTTGGTATTTGAGCAGGATTAATTGGAACTTCTCTA---AGTTTATTAATTCGGGCCGAATTAGGAAATCCTGGTTCTTTAATTGGTGAT---GACCAAATTTATAATACTATTGTAACTGCGCATGCATTTATTATAATTTTTTTCATAGTAATACCAATTTTAATTGGGGGATTTGGAAATTGATTAGTTCCACTAATA---TTAGGAGCCCCTGATATAGCTTTCCCCCGTATAAATAATATAAGTTTTTGACTTTTACCCCCCTCATTAACTTTATTAATTTCTAGAAGTATAGTTGAAAATGGAGCAGGAACAGGATGAACAGTGTACCCCCCACTCTCATCTAATATTGCTCATGGTGGAAGTTCAGTAGACTTA---GCCATTTTTTCATTACATTTAGCTGGAATTTCATCAATTTTAGGTGCAATTAATTTTATTACAACTATTATTAATATACGAAGTAATGGAATAACATTTGATCGAATACCTTTATTCGTTTGAGCTGTTGGTATTACTGCTTTACTTTTACTTTTATCACTCCCTGTTTTAGCTGGG---GCTATTACCATATTATTAACA---------------------------------------------------\\r\\n')\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import os\n",
    "\n",
    "outdir = \"micromoth_fasta/\"\n",
    "if not os.path.exists(outdir):\n",
    "    os.mkdir(outdir)\n",
    "\n",
    "fasta_file = \"micromoth.fasta\"\n",
    "with open(fasta_file) as infile:\n",
    "    dat = zip(infile, infile)\n",
    "print(dat[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "for samp, seq in dat:\n",
    "    name = samp.split(\"|\")[1].replace(\" \", \"_\")\n",
    "    with open(outdir+name+\".fasta\", 'a') as outfile:\n",
    "        outfile.write(samp)\n",
    "        outfile.write(seq)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fix alignments\n",
    "muscle writes out multi-line fasta, so we have this nasty awk script to write out single line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘micromoth_fasta_aligned’: File exists\n",
      "Acleris_hastiana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Acleris_hastiana 2 seqs, lengths min 576, max 576, avg 576\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Acleris_holmiana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Acleris_holmiana 2 seqs, lengths min 582, max 582, avg 582\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Acleris_laterana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Acleris_laterana 8 seqs, lengths min 549, max 588, avg 564\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Agonopterix_angelicella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Agonopterix_angelicella 2 seqs, lengths min 573, max 573, avg 573\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Agonopterix_arenella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Agonopterix_arenella 10 seqs, lengths min 555, max 588, avg 564\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     29 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Agonopterix_ciliella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Agonopterix_ciliella 96 seqs, lengths min 528, max 591, avg 572\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     59 MB(3%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     59 MB(3%)  Iter   1  100.00%  Root alignment\n",
      "00:00:01     59 MB(3%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:01     59 MB(3%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     59 MB(3%)  Iter   2  100.00%  Root alignment\n",
      "00:00:02     59 MB(3%)  Iter   3  100.00%  Refine biparts\n",
      "Agonopterix_heracliana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Agonopterix_heracliana 532 seqs, lengths min 519, max 606, avg 573\n",
      "00:00:01     27 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:01     27 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:04   211 MB(10%)  Iter   1  100.00%  Align node       \n",
      "00:00:04   211 MB(10%)  Iter   1  100.00%  Root alignment\n",
      "00:00:07   211 MB(10%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:07   211 MB(10%)  Iter   2  100.00%  Root alignment\n",
      "00:00:07   211 MB(10%)  Iter   2  100.00%  Root alignment\n",
      "00:00:34   211 MB(10%)  Iter   3  100.00%  Refine biparts\n",
      "Agonopterix_ocellana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Agonopterix_ocellana 2 seqs, lengths min 567, max 567, avg 567\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Agriphila_straminella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Agriphila_straminella 164 seqs, lengths min 516, max 633, avg 567\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:01     83 MB(4%)  Iter   1  100.00%  Align node       \n",
      "00:00:01     83 MB(4%)  Iter   1  100.00%  Root alignment\n",
      "00:00:02     84 MB(4%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:02     84 MB(4%)  Iter   2  100.00%  Root alignment\n",
      "00:00:02     84 MB(4%)  Iter   2  100.00%  Root alignment\n",
      "00:00:06     84 MB(4%)  Iter   3  100.00%  Refine biparts\n",
      "Alucita_hexadactyla.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Alucita_hexadactyla 4 seqs, lengths min 582, max 583, avg 582\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Ancylis_badiana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Ancylis_badiana 16 seqs, lengths min 576, max 591, avg 583\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   3  100.00%  Refine biparts\n",
      "Anthophila_fabriciana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Anthophila_fabriciana 6 seqs, lengths min 408, max 567, avg 510\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Argyresthia_abdominalis.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Argyresthia_abdominalis 2 seqs, lengths min 579, max 579, avg 579\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Argyresthia_conjugella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Argyresthia_conjugella 2 seqs, lengths min 552, max 552, avg 552\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Argyresthia_curvella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Argyresthia_curvella 6 seqs, lengths min 546, max 582, avg 565\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Argyresthia_glabratella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Argyresthia_glabratella 4 seqs, lengths min 579, max 579, avg 579\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Argyresthia_goedartella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Argyresthia_goedartella 4 seqs, lengths min 588, max 588, avg 588\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Argyresthia_pruniella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Argyresthia_pruniella 14 seqs, lengths min 549, max 594, avg 578\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     30 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     30 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     30 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     30 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Argyresthia_sorbiella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Argyresthia_sorbiella 2 seqs, lengths min 561, max 561, avg 561\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Autographa.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Autographa 2 seqs, lengths min 576, max 576, avg 576\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Bryotropha_similis.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Bryotropha_similis 4 seqs, lengths min 576, max 576, avg 576\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Bryotropha_terrella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Bryotropha_terrella 20 seqs, lengths min 549, max 591, avg 574\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     32 MB(2%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     32 MB(2%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     32 MB(2%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     32 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     32 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     32 MB(2%)  Iter   3  100.00%  Refine biparts\n",
      "Bucculatrix.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Bucculatrix 2 seqs, lengths min 555, max 555, avg 555\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Bucculatrix_nigricomella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Bucculatrix_nigricomella 14 seqs, lengths min 555, max 591, avg 581\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     30 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     30 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     30 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     30 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     30 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     30 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Caloptilia_betulicola.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Caloptilia_betulicola 8 seqs, lengths min 567, max 591, avg 578\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Caloptilia_elongella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Caloptilia_elongella 4 seqs, lengths min 546, max 576, avg 561\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Caloptilia.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Caloptilia 2 seqs, lengths min 570, max 570, avg 570\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Caloptilia_rufipennella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Caloptilia_rufipennella 8 seqs, lengths min 573, max 591, avg 586\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:00     28 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Caloptilia_stigmatella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Caloptilia_stigmatella 2 seqs, lengths min 576, max 576, avg 576\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Caryocolum_cassella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Caryocolum_cassella 2 seqs, lengths min 582, max 582, avg 582\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Caryocolum_pullatella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Caryocolum_pullatella 42 seqs, lengths min 567, max 630, avg 584\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     40 MB(2%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     40 MB(2%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     40 MB(2%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     40 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     40 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     40 MB(2%)  Iter   3  100.00%  Refine biparts\n",
      "Celypha_lacunana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Celypha_lacunana 16 seqs, lengths min 564, max 633, avg 589\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     32 MB(2%)  Iter   3  100.00%  Refine biparts\n",
      "Celypha_siderana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Celypha_siderana 16 seqs, lengths min 534, max 624, avg 574\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   3  100.00%  Refine biparts\n",
      "Cerapteryx_graminis.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Cerapteryx_graminis 134 seqs, lengths min 516, max 609, avg 572\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:01     72 MB(4%)  Iter   1  100.00%  Align node       \n",
      "00:00:01     72 MB(4%)  Iter   1  100.00%  Root alignment\n",
      "00:00:02     72 MB(4%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:02     72 MB(4%)  Iter   2  100.00%  Root alignment\n",
      "00:00:02     72 MB(4%)  Iter   2  100.00%  Root alignment\n",
      "00:00:04     72 MB(4%)  Iter   3  100.00%  Refine biparts\n",
      "Charanyca_ferruginea.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Charanyca_ferruginea 2 seqs, lengths min 585, max 585, avg 585\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Chionodes_ignorantella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Chionodes_ignorantella 2 seqs, lengths min 564, max 564, avg 564\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Cnephasia_asseclana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Cnephasia_asseclana 6 seqs, lengths min 576, max 588, avg 581\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Cochylis_nana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Cochylis_nana 2 seqs, lengths min 567, max 567, avg 567\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Coleophora_serratella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Coleophora_serratella 8 seqs, lengths min 588, max 591, avg 588\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Denisia_similella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Denisia_similella 2 seqs, lengths min 588, max 588, avg 588\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Depressaria.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Depressaria 6 seqs, lengths min 534, max 618, avg 573\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Diachrysia_chrysitis.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Diachrysia_chrysitis 4 seqs, lengths min 543, max 577, avg 560\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:00     26 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Diarsia_rubi.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Diarsia_rubi 2 seqs, lengths min 591, max 591, avg 591\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Dichrorampha.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Dichrorampha 2 seqs, lengths min 591, max 591, avg 591\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Ectropis_crepuscularia.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Ectropis_crepuscularia 2 seqs, lengths min 588, max 588, avg 588\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Elachista_adscitella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Elachista_adscitella 4 seqs, lengths min 579, max 591, avg 585\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Elachista_albifrontella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Elachista_albifrontella 88 seqs, lengths min 543, max 614, avg 583\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:01     56 MB(3%)  Iter   1  100.00%  Align node       \n",
      "00:00:01     56 MB(3%)  Iter   1  100.00%  Root alignment\n",
      "00:00:01     57 MB(3%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:01     57 MB(3%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     57 MB(3%)  Iter   2  100.00%  Root alignment\n",
      "00:00:02     57 MB(3%)  Iter   3  100.00%  Refine biparts\n",
      "Elachista_apicipunctella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Elachista_apicipunctella 12 seqs, lengths min 516, max 591, avg 571\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:01     29 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:01     29 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     29 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     29 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Elachista_canapennella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Elachista_canapennella 6 seqs, lengths min 555, max 564, avg 558\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Elachista_humilis.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Elachista_humilis 4 seqs, lengths min 555, max 591, avg 573\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Epermenia_chaerophyllella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Epermenia_chaerophyllella 16 seqs, lengths min 528, max 594, avg 570\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   3  100.00%  Refine biparts\n",
      "Epinotia_nanana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Epinotia_nanana 14 seqs, lengths min 537, max 588, avg 572\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     30 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     30 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     30 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     30 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Epinotia_signatana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Epinotia_signatana 2 seqs, lengths min 591, max 591, avg 591\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Epinotia_tedella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Epinotia_tedella 28 seqs, lengths min 546, max 592, avg 583\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     35 MB(2%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     35 MB(2%)  Iter   1  100.00%  Root alignment\n",
      "00:00:01     35 MB(2%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:01     35 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     35 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     35 MB(2%)  Iter   3  100.00%  Refine biparts\n",
      "Epinotia_tenerana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Epinotia_tenerana 16 seqs, lengths min 525, max 610, avg 573\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     31 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   3  100.00%  Refine biparts\n",
      "Eriocrania_salopiella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Eriocrania_salopiella 2 seqs, lengths min 576, max 576, avg 576\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Eriocrania_semipurpurella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Eriocrania_semipurpurella 4 seqs, lengths min 591, max 600, avg 595\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Euhyponomeutoides_albithoracellus.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Euhyponomeutoides_albithoracell 42 seqs, lengths min 537, max 594, avg 582\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:01     40 MB(2%)  Iter   1  100.00%  Align node       \n",
      "00:00:01     40 MB(2%)  Iter   1  100.00%  Root alignment\n",
      "00:00:01     40 MB(2%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:01     40 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     40 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     40 MB(2%)  Iter   3  100.00%  Refine biparts\n",
      "Euhyponomeutoides_ribesiella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Euhyponomeutoides_ribesiella 2 seqs, lengths min 588, max 588, avg 588\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Eulithis_prunata.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Eulithis_prunata 12 seqs, lengths min 531, max 600, avg 572\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     29 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Eupithecia_subfuscata.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Eupithecia_subfuscata 6 seqs, lengths min 549, max 579, avg 562\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Euspilapteryx_auroguttella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Euspilapteryx_auroguttella 2 seqs, lengths min 588, max 588, avg 588\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Exapate_congelatella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Exapate_congelatella 2 seqs, lengths min 585, max 585, avg 585\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Gillmeria_pallidactyla.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Gillmeria_pallidactyla 4 seqs, lengths min 485, max 585, avg 535\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Glyphipterix_bergstraesserella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Glyphipterix_bergstraesserella 2 seqs, lengths min 555, max 555, avg 555\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Glyphipterix_simpliciella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Glyphipterix_simpliciella 210 seqs, lengths min 525, max 621, avg 580\n",
      "00:00:00     24 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     24 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:01    100 MB(5%)  Iter   1  100.00%  Align node       \n",
      "00:00:01    100 MB(5%)  Iter   1  100.00%  Root alignment\n",
      "00:00:02    100 MB(5%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:02    100 MB(5%)  Iter   2  100.00%  Root alignment\n",
      "00:00:02    100 MB(5%)  Iter   2  100.00%  Root alignment\n",
      "00:00:07    100 MB(5%)  Iter   3  100.00%  Refine biparts\n",
      "Gracillaria_syringella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Gracillaria_syringella 8 seqs, lengths min 529, max 576, avg 562\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     28 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Hylaea_fasciaria.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Hylaea_fasciaria 2 seqs, lengths min 592, max 592, avg 592\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Hypena_proboscidalis.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Hypena_proboscidalis 8 seqs, lengths min 522, max 579, avg 561\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Lampronia_capitella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Lampronia_capitella 6 seqs, lengths min 576, max 591, avg 586\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Lampronia_corticella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Lampronia_corticella 4 seqs, lengths min 579, max 582, avg 580\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Lampronia_flavimitrella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Lampronia_flavimitrella 2 seqs, lengths min 591, max 591, avg 591\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Lepidoptera.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Lepidoptera 2 seqs, lengths min 513, max 513, avg 513\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Lyonetia_clerkella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Lyonetia_clerkella 204 seqs, lengths min 513, max 624, avg 575\n",
      "00:00:00     24 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     24 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:01     98 MB(5%)  Iter   1  100.00%  Align node       \n",
      "00:00:01     98 MB(5%)  Iter   1  100.00%  Root alignment\n",
      "00:00:02     98 MB(5%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:02     98 MB(5%)  Iter   2  100.00%  Root alignment\n",
      "00:00:02     98 MB(5%)  Iter   2  100.00%  Root alignment\n",
      "00:00:08     98 MB(5%)  Iter   3  100.00%  Refine biparts\n",
      "Macaria_wauaria.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Macaria_wauaria 6 seqs, lengths min 577, max 591, avg 585\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Martania_taeniata.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Martania_taeniata 6 seqs, lengths min 579, max 591, avg 587\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Mompha_langiella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Mompha_langiella 88 seqs, lengths min 546, max 591, avg 574\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     56 MB(3%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     56 MB(3%)  Iter   1  100.00%  Root alignment\n",
      "00:00:01     56 MB(3%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:01     56 MB(3%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     56 MB(3%)  Iter   2  100.00%  Root alignment\n",
      "00:00:02     56 MB(3%)  Iter   3  100.00%  Refine biparts\n",
      "Notocelia_cynosbatella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Notocelia_cynosbatella 2 seqs, lengths min 582, max 582, avg 582\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Ocnerostoma_friesei.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Ocnerostoma_friesei 2 seqs, lengths min 588, max 588, avg 588\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Ocnerostoma_piniariella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Ocnerostoma_piniariella 6 seqs, lengths min 556, max 589, avg 577\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Odontopera_bidentata.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Odontopera_bidentata 4 seqs, lengths min 582, max 585, avg 583\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:01     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:01     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:01     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Orthosia_gothica.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Orthosia_gothica 12 seqs, lengths min 555, max 591, avg 579\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     29 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Pharmacis_fusconebulosa.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Pharmacis_fusconebulosa 4 seqs, lengths min 498, max 594, avg 546\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Phyllocnistis.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Phyllocnistis 2 seqs, lengths min 513, max 513, avg 513\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Phyllocnistis_labyrinthella.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Phyllocnistis_labyrinthella 360 seqs, lengths min 507, max 621, avg 577\n",
      "00:00:00     25 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     25 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:03    153 MB(8%)  Iter   1  100.00%  Align node       \n",
      "00:00:03    153 MB(8%)  Iter   1  100.00%  Root alignment\n",
      "00:00:05    154 MB(8%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:05    154 MB(8%)  Iter   2  100.00%  Root alignment\n",
      "00:00:05    154 MB(8%)  Iter   2  100.00%  Root alignment\n",
      "00:00:18    154 MB(8%)  Iter   3  100.00%  Refine biparts\n",
      "Phyllonorycter_emberizaepenella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Phyllonorycter_emberizaepenella 8 seqs, lengths min 552, max 588, avg 578\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     28 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     28 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Phyllonorycter_strigulatella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Phyllonorycter_strigulatella 6 seqs, lengths min 537, max 591, avg 555\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:01     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:01     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:01     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Plutella_xylostella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Plutella_xylostella 196 seqs, lengths min 513, max 603, avg 575\n",
      "00:00:00     24 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     24 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:01     94 MB(5%)  Iter   1  100.00%  Align node       \n",
      "00:00:01     94 MB(5%)  Iter   1  100.00%  Root alignment\n",
      "00:00:02     95 MB(5%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:02     95 MB(5%)  Iter   2  100.00%  Root alignment\n",
      "00:00:02     95 MB(5%)  Iter   2  100.00%  Root alignment\n",
      "00:00:07     95 MB(5%)  Iter   3  100.00%  Refine biparts\n",
      "00:00:11     95 MB(5%)  Iter   4  100.00%  Refine biparts\n",
      "Scotopteryx_chenopodiata.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Scotopteryx_chenopodiata 2 seqs, lengths min 564, max 564, avg 564\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Selenia_dentaria.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Selenia_dentaria 2 seqs, lengths min 564, max 564, avg 564\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Stigmella_aeneofasciella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Stigmella_aeneofasciella 2 seqs, lengths min 561, max 561, avg 561\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Stigmella_lapponica.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Stigmella_lapponica 4 seqs, lengths min 549, max 561, avg 555\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     26 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Stigmella_magdalenae.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Stigmella_magdalenae 10 seqs, lengths min 564, max 612, avg 586\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:01     29 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Stigmella_sorbi.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Stigmella_sorbi 10 seqs, lengths min 579, max 588, avg 585\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     29 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     29 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     29 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Syndemis_musculana.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Syndemis_musculana 6 seqs, lengths min 582, max 591, avg 588\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Thera.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Thera 2 seqs, lengths min 591, max 591, avg 591\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Thera_juniperata.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Thera_juniperata 2 seqs, lengths min 558, max 558, avg 558\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Udea_lutealis.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Udea_lutealis 16 seqs, lengths min 510, max 592, avg 565\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     31 MB(2%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   2  100.00%  Refine tree   \n",
      "00:00:00     31 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     31 MB(2%)  Iter   3  100.00%  Refine biparts\n",
      "Xanthorhoe_montanata.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Xanthorhoe_montanata 6 seqs, lengths min 582, max 591, avg 585\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     27 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   2  100.00%  Root alignment\n",
      "00:00:00     27 MB(1%)  Iter   3  100.00%  Refine biparts\n",
      "Xestia_baja.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Xestia_baja 2 seqs, lengths min 588, max 588, avg 588\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n",
      "Ypsolopha_dentella.fasta\n",
      "\n",
      "MUSCLE v3.8.1551 by Robert C. Edgar\n",
      "\n",
      "http://www.drive5.com/muscle\n",
      "This software is donated to the public domain.\n",
      "Please cite: Edgar, R.C. Nucleic Acids Res 32(5), 1792-97.\n",
      "\n",
      "Ypsolopha_dentella 2 seqs, lengths min 540, max 540, avg 540\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 1\n",
      "00:00:00     23 MB(1%)  Iter   1  100.00%  K-mer dist pass 2\n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Align node       \n",
      "00:00:00     26 MB(1%)  Iter   1  100.00%  Root alignment\n"
     ]
    }
   ],
   "source": [
    "!mkdir micromoth_fasta_aligned\n",
    "!for f in `ls micromoth_fasta`; do echo $f; muscle --quiet -in micromoth_fasta/$f -out micromoth_fasta_aligned/$f; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd micromoth_fasta_aligned\n",
    "for i in `ls`; do echo $i;  awk '{if(NR==1) {print $0} else {if($0 ~ /^>/) {print \"\\n\"$0} else {printf $0}}}' $i > tmp.fasta; mv tmp.fasta $i; done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "!../make_obs.py -f micromoth_fasta_aligned/ -o micromoth.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tbin_0\tbin_1\tbin_2\tbin_3\tbin_4\tbin_5\tbin_6\tbin_7\tbin_8\tbin_9\r\n",
      "59\t15\t14\t3\t5\t1\t2\t1\t1\t1"
     ]
    }
   ],
   "source": [
    "cat micromoth.obs"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
